src/sentry/snuba/metrics/extraction.py

```diff
@@ -1050,12 +1077,12 @@ def __init__(
         environment: Optional[str] = None,
         groupbys: Optional[Sequence[str]] = None,
         spec_type: MetricSpecType = MetricSpecType.SIMPLE_QUERY,
-        spec_version: Optional[dict[str, bool]] = None,
+        spec_version: Optional[SpecVersion] = None,
     ):
         self.field = field
         self.query = query
         self.spec_type = spec_type
-        self.spec_version = spec_version if spec_version else SPEC_VERSIONS[0]
+        self.spec_version = spec_version if spec_version else SPEC_VERSIONS[1]
```

Comment:
`SPEC_VERSIONS[1]` is the one that contains the True value for `use_updated_env_logic`.
Perhaps I should add here `get_spec("use_updated_env_logic")` to be more explicit.